packages/knip/src/plugins/knex/index.ts

```diff
@@ -0,0 +1,69 @@
+import type { IsPluginEnabled, Plugin, ResolveConfig, ResolveFromAST } from '../../types/config.js';
+import type ts from 'typescript';
+import { type Input, toDependency, toEntry } from '../../util/input.js';
+import { hasDependency } from '../../util/plugin.js';
+import type { KnexConfig } from './types.js';
+import { clientToPackages, getKnexClients } from './helpers.js';
+
+// https://knexjs.org
+
+const title = 'Knex';
+
+const enablers = ['knex'];
+
+const isEnabled: IsPluginEnabled = ({ dependencies }) => hasDependency(dependencies, enablers);
+
+const config = ['knexfile.{js,cjs,mjs,ts,cts,mts}'];
+
+const processKnexConfig = (config: KnexConfig): Input[] => {
+  const inputs: Input[] = [];
+  if (config.client) {
+    const packages = clientToPackages(config.client);
+    inputs.push(...packages.map(pkg => toDependency(pkg, { optional: true })));
+  }
+  if (config.migrations?.directory) {
+    const dirs = Array.isArray(config.migrations.directory)
+      ? config.migrations.directory
+      : [config.migrations.directory];
+    inputs.push(...dirs.map(dir => toEntry(`${dir}/*.{js,ts}`)));
+  }
+  if (config.seeds?.directory) {
+    const dirs = Array.isArray(config.seeds.directory) ? config.seeds.directory : [config.seeds.directory];
+    inputs.push(...dirs.map(dir => toEntry(`${dir}/*.{js,ts}`)));
+  }
+  return inputs;
```

Comment:
This helper is great. Would be great to actually have a migration and a seed file just to show they won't end up unused.